lib/consumer/data_updater.ex

```diff
@@ -116,25 +116,48 @@ defmodule Mississippi.Consumer.DataUpdater do
            timestamp,
            state.handler_state
          ) do
-      {:ok, _, new_handler_state} ->
-        _ = Logger.debug("Successfully handled message #{inspect(meta.message_id)}")
-        {:ok, message_tracker} = MessageTracker.get_message_tracker(state.sharding_key)
-        MessageTracker.ack_delivery(message_tracker, message)
+      {:ack, _, new_handler_state} ->
+        ack_message!(message, state.sharding_key)
 
         new_state = %State{state | handler_state: new_handler_state}
 
         {:noreply, new_state, @data_updater_deactivation_interval_ms}
 
-      {:error, reason, _state} ->
-        _ =
-          Logger.warning("Error handling message #{inspect(meta.message_id)}, reason #{inspect(reason)}")
+      {:ack, _, new_handler_state, {:continue, continue_arg}} ->
+        ack_message!(message, state.sharding_key)
+
+        new_state = %State{state | handler_state: new_handler_state}
 
-        {:ok, message_tracker} = MessageTracker.get_message_tracker(state.sharding_key)
-        MessageTracker.reject(message_tracker, message)
+        {:noreply, new_state, {:continue, continue_arg}}
+
+      {:discard, reason, _state} ->
+        reject_message!(message, state.sharding_key, reason)
         {:noreply, state, @data_updater_deactivation_interval_ms}
+
+      {:discard, reason, _state, {:continue, continue_arg}} ->
+        reject_message!(message, state.sharding_key, reason)
+        {:noreply, state, {:continue, continue_arg}}
```

Comment:
although we are discarding the message we should still update the handler state with the updated value from the message handle